### PR TITLE
Fixes #9418: Defined method 'notes_for_tags' in user.rb

### LINKED ISSUE #9418: NoMethodError: undefined method `notes_for_tags' for #<User:0x00007f95d21db590>
Sentry Issue: [PLOTS2-18A](https://sentry.io/organizations/publiclab/issues/2310388476/?referrer=github_integration)

```
NoMethodError: undefined method `notes_for_tags' for #<User:0x00007f95d21db590>
Did you mean?  note_streak
  app/controllers/notes_controller.rb:336:in `author_topic'
    @notes = @user.notes_for_tags(@tagnames)
...
(104 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by KarishmaVanwari**
Fixes #9418 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -544,4 +544,14 @@ def map_openid_registration(registration)
...
+
+  def notes_for_tags(tagnames)
+    nids = NodeTag.includes(:node, :tag)
+    .references(:term_data)
+    .where(type: 'note')
+    .where(name: tagnames, uid: :uid)
+    @pagy, @notes = pagy(Node
+      .order('nid DESC')
+      .where(nid: nids), items: 24)
+  end
```
Comment (TildaDares) [inline app/models/user.rb:556]: @KarishmaVanwari You added the method to the private methods.

Also, I'm not sure this code will work. I played around with it but it's not returning what it's supposed to.

Can you test this code and see if it works properly?
```
def notes_for_tags(tagnames)
    Node.includes(:node_tag, :tag)
    .where('term_data.name IN (?)', tagnames)
    .references(:term_data, :node_tag)
    .where('node.uid = ?', uid)
    .where(type: 'note')
    .order('node.nid DESC')
  end
```

Comment (KarishmaVanwari) [reply app/models/user.rb:556]: Hey @TildaDares, could you please brief me as to how do I check if the given piece of code works well?

Comment (TildaDares) [reply app/models/user.rb:556]: @KarishmaVanwari Have you able to test this?

Comment (KarishmaVanwari) [reply app/models/user.rb:556]: I'm on it @TildaDares. Apologies for the delay since I had been busy with my university exams.

**Comment by TildaDares** [comment]
Resolved in #11474

**Pull request closed without merging**